Description: Bug#1091381:xtail FTBFS on 32-bit architectures due to 64bit time_t
Since we turned on 64-bit time_t, xtail now fails to build on most 32-bit
architectures.  The attached trivial patch fixes the build.

[PATCH] Fix build failure on 32-bit platforms

Use the time_t type instead of the hardcoded long type for mtime variable.
This is necessary since we recently enabled 64-bit time_t on most 32-bit
platforms.

Signed-off-by: Helge Deller <deller@gmx.de>

Bug-Debian: #1091381
Forwarded: not-needed
Origin: Helge Deller <deller@gmx.de>
Last-Update: 2024-12-25

---

diff -up ./xtail.h.org ./xtail.h
--- xtail.h.org	2024-12-25 00:31:31.133262689 +0000
+++ xtail.h	2024-12-25 00:31:50.029631603 +0000
@@ -108,7 +108,7 @@ struct entry_descrip {
     char *name;		/* pathname to the entry			*/
     int fd;		/* opened fd, or <= 0 if not opened		*/
     long size;		/* size of entry last time checked		*/
-    long mtime;		/* modification time last time checked		*/
+    time_t mtime;	/* modification time last time checked		*/
     dev_t dev;		/* device number last time, if file is opened	*/
     ino_t ino;		/* inode number last time, if file is opened	*/
 };

